Iter1

jbakhtin opened the pull request.

Commit:
Init commit

jbakhtin committed:
Update README.md

jbakhtin committed:
Update README.md

jbakhtin committed:
Update README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
create docs/

jbakhtin committed:
create diagram without name.drawio.png

Commit:
Changes:
- rename image name from cyrillic to latin
- add db structure image to README.md

Commit:
update README.md

Commit:
delete .github/workflows/gophermart.yml file

Commit:
Changes:
- added handlers

Commit:
update README.md

jbakhtin committed:
update db-structure.drawio.png

Commit:
update README.md

jbakhtin committed:
Update db-structure.drawio.png

Commit:
Features:
- added migrations for for users table
- added handlers
- added server
- added connection to db

Commit:
update README.md

Commit:
Update README.md

Commit:
change GetById to GetByID

Commit:
change Id to ID

Commit:
set status code into handler mark Update()

mr-fard4y commented on cmd/routecoverage/main.go:
```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"context"
```
импорты

mr-fard4y commented on README.md:
```diff
@@ -1,25 +1,111 @@
+
+### Структура БД
+
+![DB Structure](docs/images/db-structure.drawio.png)
```
1) нет описания таблиц - кртакое описание, назначение и структура
2) для всех таблиц, кроме users нужно поле description, описывающее доп.инфу про запись

mr-fard4y replied:
перерисовать, чтобы количество линий и их расположение было более читабельным:
1 колонца - Users
2 колонка - Routes, Areas
3 колонка - Marks, Points

mr-fard4y replied:
при необходимости хранения активности пользователя, Users можно разделить на 2 таблицы:
1) общая информация о пользователе - имя, почта, etc
2) авторизационная информация о пользователей - логин, пароль, временные-метки активности

mr-fard4y commented on internal/server/server.go:
```diff
@@ -0,0 +1,130 @@
+	// TODO: добавить проверку авторизацию
+
+	// handlers
+	authHandler, err := auth.NewHandler(*s.config)
```
для чего здесь 5 "одинаковых" хэндлеров?

в рамках пакета описать один, и лоигку взаимодействия разбить по файлам, НО не отдельным пакетам

mr-fard4y commented on internal/storages/sqlstorage/entities/area/area.go:
```diff
@@ -0,0 +1,44 @@
+	"go.uber.org/zap"
+)
+
+type Storage struct {
```
для чего каждая модель данных содержит свою структуру Storage?

mr-fard4y commented on internal/storages/sqlstorage/migrations/20230417223555_create_users_table.sql:
```diff
@@ -0,0 +1,15 @@
+                      id bigserial NOT NULL PRIMARY KEY,
+                      name text NOT NULL,
+                      lastname text NOT NULL,
+                      login text NOT NULL UNIQUE,
```
логин и иные приватные данные должны храниться в шифрованном виде, а не обычном тексте

[pgcrypto](https://www.postgresql.org/docs/12/pgcrypto.html#id-1.11.7.34.6)

jbakhtin replied:
Они будут хэшироваться на уровне приложение, не на уровне sql

mr-fard4y commented on internal/storages/sqlstorage/migrations/20230417223555_create_users_table.sql:
```diff
@@ -0,0 +1,15 @@
+                      login text NOT NULL UNIQUE,
+                      email text NOT NULL UNIQUE,
+                      password text NOT NULL,
+                      created_at text NOT NULL,
```
время хранится в таймстампах или нужной типе времени

[datetime](https://postgrespro.ru/docs/postgresql/9.4/datatype-datetime)

jbakhtin replied:
Да, мой косяк, проглядел, на дальнейших стадиях разработки всплыло бы

mr-fard4y commented on README.md:
```diff
@@ -1,25 +1,111 @@
+
 
-Затем добавьте полученные изменения в свой репозиторий.
+### Инкремент 2
```
нет фоновых задач и задач с конкурентностью/многопоточностью

jbakhtin replied:
В фоновом процессе будет отправка уведомлений, на данный момент пока что уведомления будут типа "подтвердить почту"
Других кейсов пока не придумал в рамках поставленной задачи

mr-fard4y commented on README.md:
```diff
@@ -1,25 +1,111 @@
+
+### Структура БД
+
+![DB Structure](docs/images/db-structure.drawio.png)
```
для чего и почему все таблицы завязаны на пользователя?
продумать и упростить связи между таблицами

jbakhtin replied:
Упростил немного схему:
Например принадлежность маршрутов к заданной территории на карте будет определяться PostGIS на основе географических данных
Остальные же сущности это создаваемые пользователем объекты на карте, которые должны быть привязаны к нему

Commit:
Changes:
- changed structure of progect
- added diy auth with redis

Commit:
Features:
- added jwt auth
- register
- login
- check auth middleware

Commit:
Changed:
- auth with jwt-go

Commit:
Merge branch 'iter-1-auth' into iter1

Commit:
Changes:
- added error handlers

Commit:
Refactoring:
- changes related with style

jbakhtin committed:
Update db-structure.drawio.png

Commit:
Changes:
- goimports

Commit:
Features:
- use config to auth

tmvrus reviewed (changes requested):
Где тесты?

Commit:
Features:
- added test for auth handler (Registration and Autentification)

Commit:
Changes:
- added tests for auth handler
- changed apperror handler
- changed design of routes

Commit:
Changes:
- closed response body in test file

Commit:
Changes:
- added req.Body.Close() for tests

Commit:
Changes:
- changde tests

Commit:
Updated README.md

tmvrus reviewed (changes requested):
Из заявленного сделаны только хендлеры для пользователя, такую работу я принять не могу

Commit:
Features:
- added email worker pool
- bility to send messages

Commit:
Updated README.md
